sdk/monitor/Azure.Monitor.OpenTelemetry.LiveMetrics/src/Internals/Manager.State.cs

```diff
@@ -117,11 +119,11 @@ private void Run(CancellationToken cancellationToken)
                 else
                 {
                     // Subtract the time spent in this tick when scheduling the next tick so that the average period is close to the intended.
-                    nextTick = _period - timeSpentInThisTick;
+                    nextTick = _period - timeSpentInThisCallback;
                     nextTick = nextTick > TimeSpan.Zero ? nextTick : TimeSpan.Zero;
                 }
 
-                Task.Delay(nextTick, cancellationToken).Wait();
+                _thread!.Join(nextTick);
```

Comment:
The `Thread.Join` method is typically used to block the calling thread until the thread that Join is called on terminates. However, in this context, it's being called within the thread's own execution loop. This is a mistake, as a thread cannot join itself. It would cause the thread to wait indefinitely for itself to finish, which is a logical deadlock.

Reply:
> It would cause the thread to wait indefinitely for itself to finish

I'm having a misunderstanding because this indefinite wait isn't happening. I tested this today and it was working as expected.

My understanding, 
- the initial application thread calls `_thread.Start()`. This was the first calling thread.
- When the `Run` method is invoked, it runs the primary `while(true)` loop. At this point it is now the calling thread. At the end of the loop, it calls `_thread.Join(timeout)`. When that timeout is exceeded, it's returning execution to the calling thread, which is the primary while loop.

I can spend more time Thursday digging into this.